### PR TITLE
Add global image registry to helm chart

### PR DESCRIPTION
**Pull request by Omri-sh**
Add global image registry to helm chart.
Taking inspiration from bitnami's helm-charts

### CODE REVIEW

**Comment by weeco** [comment]
Hi @Omri-sh ,
this would be a breaking change. What is the advantage/usecases of this change?

Since there's just one image used in this chart I don't see the advantage in contrast to the current solution where you can change the `image.repository` value to whatever registry / repo you want as well.

**Comment by yardenshoham** [comment]
Hey @weeco, think about a big helm chart that has tens of subcharts. It would be a lot of work to set each chart's `image.repository` when you can simply set `global.imageRegistry`. This is an accepted practice followed by Bitnami, the most popular helm chart repository.

**Comment by weeco** [comment]
@yardenshoham That makes sense to me for big charts, but since this is not a big chart (and most likely will never contain more than one deployment) I don't see the benefit here. I'm fine with following standards and practices, however not at the cost of breaking changes that barely provide an advantage. This chart sticks mostly to the default chart as created by the Helm CLI.

**Comment by yardenshoham** [comment]
The thing is, when this chart is used **as a subchart** it will have to be treated differently. Say I have an umbrella helm chart with the following structure:
```
umbrella
|
|-  bitnami-postgresql
|-  bitnami-kafka
|-  bitnami-spark
|-  bitnami-elasticsearch
|-  bitnami-mongodb
|-  cloudhut-kowl
```

If I want all helm charts to use my private image registry `docker.mydomain.com`, all I need to do is set,
```yaml
global:
  imageRegistry: docker.mydomain.com
```
and run `helm install umbrella .`.

But with `kowl` included (with its configuration before this PR), my `values.yaml` would become,
```yaml
global:
  imageRegistry: docker.mydomain.com

cloudhut-kowl:
  image:
    repository: docker.mydomain.com/cloudhut/kowl
```

Do you see the issue?

**Comment by weeco** [comment]
Got you, is Kowl currently part of such an umbrella chart?

**Comment by yardenshoham** [comment]
We would like to add it. If this PR is merged we will add it.

File: README.md [hunk 1]
```diff
@@ -15,7 +15,9 @@ helm install -f values.yaml --name=kowl cloudhut/kowl
 | Parameter | Description | Default |
 | --- | --- | --- |
 | `replicaCount` | Number of Kowl replicas | `1` |
-| `image.repository` | Docker image repo | `quay.io/cloudhut/kowl` |
+| `global.imageRegistry` | Global Image Registry |` ` |
+| `image.registry` | Image Registry |`quay.io` |
```
Comment (weeco) [inline README.md:19]:
```suggestion
| `global.imageRegistry` | Global Image Registry | (none) |
| `image.registry` | Image Registry | `quay.io` |
```

**Comment by weeco** [comment]
Makes sense. Given that most users probably stick to the default repository / images built by us, I tend to accept the PR. Will be released in a major release afterwards

**Comment by Omri-sh** [comment]
Accepted your changes.
I think this PR is ready to merged.

**Pull request merged**